app/gui/src/project-view/components/visualizations/WarningsVisualization.vue

```diff
@@ -20,17 +20,45 @@ const removeWarnings = computed(() =>
 </script>
 
 <script setup lang="ts">
-type Data = string[]
+type Error = {
+  type?: string
+  content?: {
+    argument_name?: string
+    call_location?: string
+    constructor?: string
+    function_name?: string
+    type?: string
+  }
+  message?: string
+}
+
+type Warning = string[]
+
+type Data = Error | Warning
 
 const props = defineProps<{ data: Data }>()
+const messages = ref<string[]>([])
+const isRemoveWarningsDisabled = ref<boolean>(false)
 
 const config = useVisualizationConfig()
 
+watchEffect(() => {
```

Comment:
Each change to `props.data` may add to `messages`, but entries are never removed. If `messages` should depend only on the current inputs, defining it as a `computed` expresses (and enforces) this.